### DISCUSSION TITLE
fd with multiple patterns

### DISCUSSION THREAD

**Discussion by vbd** [sxyazi/yazi#1606] [1. Q&A]
I'm on Windows with yazi 0.3.
After hitting `s` to search via fd I'm trying to find files with pattern1 and pattern2 by typing `pattern1 --and pattern2`.
But I don't get any result. Trying `fd pattern1 --and pattern2` in powershell works.
I've found https://github.com/sxyazi/yazi/pull/1013 but I'm not sure how to use it and if it is a solution to my task.
If I run it in shell (block) with `fd pattern1 --and pattern2` the screen flickers. When I quit yazi I see the result.
How can I make it work?

**Comment by xfzv** [comment]
> After hitting s to search via fd

You shouldn't use `s` in this case because it's calling `fd` without any argument (unless you've changed the default):

https://github.com/sxyazi/yazi/blob/main/yazi-config/preset/keymap.toml#L77

> I've found https://github.com/sxyazi/yazi/pull/1013 but I'm not sure how to use it and if it is a solution to my task.

I think #1013 can be useful to add some `fd` and `rg` flags to `keymap.toml`, e.g.

```toml
keymap = [
  { on = "s", run = "search fd --args --hidden", desc = "Search files by name using fd" },
  { on = "S", run = "search rg --args --hidden", desc = "Search files by content using ripgrep" },
]
```

> If I run it in shell (block) with fd pattern1 --and pattern2 the screen flickers. When I quit yazi I see the result.

Same here, it seems that `--block` doesn't work.

**Comment by LvanderGoten** [comment]
Is it possible to somehow use arguments? Like `fd -e zip` to list only zip files?

**Comment by vbd** [comment]
Today I discovered `Z` which will be enough for my use-case.
